modules/custom/social_graphql/src/GraphQL/EntityConnection.php

```diff
@@ -234,10 +237,10 @@ protected function hasResult() : bool {
    *   The result for this connection's query.
    */
   protected function getResult() : SyncPromise {
-    if (!$this->hasResult()) {
-      $this->result = $this->execute();
+    if ($this->result instanceof SyncPromise) {
+      return $this->result;
     }
-    return $this->result;
+    return $this->execute();
```

Comment:
🚨  This change breaks the code.

I'm also unclear on why the change is needed.